guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/MockInstanceImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.guvnor.m2repo.backend.server;
+
+import java.lang.annotation.Annotation;
+import java.util.Iterator;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.util.TypeLiteral;
+
+import com.google.common.collect.Iterators;
+
+public class MockInstanceImpl<T> implements Instance<T> {
```

Comment:
@adrielparedes hehe.. well, since it's possible any of our code from Uberfire up to the KIE Workbenches _could_ need a ```MockInstance``` I guess putting it [here](https://github.com/AppFormer/uberfire/tree/master/uberfire-testing-utils/src/main/java/org/uberfire/mocks) is probably best.